src/main/java/com/github/gkttk/sixth/bookdao/Main.java

```diff
@@ -1,7 +1,71 @@
 package com.github.gkttk.sixth.bookdao;
 
+import com.github.gkttk.sixth.bookdao.dao.BookDao;
+import com.github.gkttk.sixth.bookdao.dao.enums.Criteria;
+import com.github.gkttk.sixth.bookdao.dao.initializer.BookDaoInitializer;
+import com.github.gkttk.sixth.bookdao.dao.initializer.BookStorageDaoInitializer;
+import com.github.gkttk.sixth.bookdao.exception.SpecificationFactoryException;
+import com.github.gkttk.sixth.bookdao.model.Book;
+import com.github.gkttk.sixth.bookdao.view.ResultPrinter;
+import com.github.gkttk.sixth.bookdao.view.factory.ConsoleResultPrinterFactory;
+import com.github.gkttk.sixth.bookdao.view.factory.ResultPrinterFactory;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+
 public class Main {
+
+    static {
+        ResultPrinterFactory resultPrinterFactory = new ConsoleResultPrinterFactory();
+        RESULT_PRINTER = resultPrinterFactory.createResultPrinter();
+
+        BookDaoInitializer bookDAOInitializer = new BookStorageDaoInitializer();
+        BOOK_DAO = bookDAOInitializer.initializeBookDao();
+    }
+
+    private static final Logger LOGGER = LogManager.getLogger(Main.class);
+    private static final ResultPrinter RESULT_PRINTER;
+    private static final BookDao BOOK_DAO;
+
+
     public static void main(String[] args) {
 
+        try {
```

Comment:
I would not write such method, check via test, if you want to actually run, write director for one case e.g. search by authror, and test this director as well